### ISSUE TITLE
Can't writeBin without removing filename attribute when using a file parameter

### ISSUE THREAD

**Issue by meztez** [rstudio/plumber#582]
writeBin does not like the raw vector to have attributes (in this case filename)

```R
#* @apiTitle Upload test

#* @param pdf:file
#* @post /upload
function(pdf) {
  f <- tempfile()
  on.exit(unlink(f), add = TRUE)
  writeBin(pdf, tempfile())
  class(pdf)
}

#* @param pdf:file
#* @post /upload2
function(pdf) {
  f <- tempfile()
  on.exit(unlink(f), add = TRUE)
  writeBin(as.raw(pdf), tempfile())
  class(pdf)
}

#* @param pdf:file
#* @post /upload3
function(pdf) {
  f <- tempfile()
  on.exit(unlink(f), add = TRUE)
  attr(pdf, "filename") <- NULL
  writeBin(pdf, tempfile())
  class(pdf)
}

```

upload : Error in writeBin(pdf, tempfile()): can only write vector objects
upload2 and upload3 : ok

Should we fix or document use case?

If fix:
  a) remove attributes
  b) return a named list of bytes list(filename = bytesVector); comes with testing what filename constitute a valid R name.
  c) other option that comes down to where does filename should live during execution

**Comment by schloerke** [comment]
I like the solution of `/upload2`. I also like the idea of `list(filename = rawVec)` as it extends to multi-file upload as well.  If no name is provided, then we can do `list(rawVec)`.

(** Asking team)

**Comment by schloerke** [comment]
`list(filename = bytesVector)` feels so much cleaner. Waiting for more feedback.

**Comment by meztez** [comment]
PR incoming

**Comment by schloerke** [comment]
Thank you @meztez . I think the `list(filename = bytesVector)` is the better approach. 

This will make multipart forms much easier as well. (`list(f1 = val1, f2 = val2)`)

(Great catch!)

**Comment by schloerke** [comment]
Fixed in #636
